RFCs/0026-DataCollector Extensibility - Getting testSources.md

```diff
@@ -0,0 +1,55 @@
+# 0026 DataCollector Extensibility - Getting testSources in property bag 
+
+## Summary
+Getting `testSources` as part of property bag. List of test sources can be used by datacollectors for processing before test run start.
+
+## Motivation
+Data collector might need test sources list while initialization. Example: Static code coverage data collector needs to instrument the test sources before test run start.
+
+The datacollector will receive a property bag in sessionStartEventArgs
+
+```csharp
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.DataCollection;
+
+[DataCollectorFriendlyName("NewDataCollector")]
+[DataCollectorTypeUri("my://new/datacollector")]
+public class NewDataCollector : DataCollector
+{
+    private string logFileName;
+	private IEnumerable<string> testSources;
+    private DataCollectionEnvironmentContext context;
+
+    public override void Initialize(
+            System.Xml.XmlElement configurationElement,
+            DataCollectionEvents events,
+            DataCollectionSink dataSink,
+            DataCollectionLogger logger,
+            DataCollectionEnvironmentContext environmentContext)
+    {
+        events.SessionStart += this.SessionStarted_Handler;
+        events.TestCaseStart += this.Events_TestCaseStart;
+        this.logFileName = configurationElement["LogFileName"];
+		this.context = environmentContext;
+    }
+    
+    private void SessionStarted_Handler(object sender, SessionStartEventArgs sessionStartEventArgs)
+    {
+        var filename = Path.Combine(AppContext.BaseDirectory, logFileName);
```

Comment:
I would like to understand pros and cons with other approaches we had compared to this approach